src/audio.rs

```diff
@@ -1087,37 +1085,48 @@ pub fn read_audio_packet(ident :&IdentHeader, setup :&SetupHeader, packet :&[u8]
 		assert_eq!(audio_spectri.len(), prev_data.len());
 
 		let win_slope = &ident.cached_bs_derived[left_n_use_bs1 as usize].window_slope;
-		for (prev_chan, chan) in
-				prev_data.iter().zip(audio_spectri.iter_mut()) {
+
+		for (prev_chan, chan) in prev_data.into_iter().zip(audio_spectri.iter_mut()) {
 			let plen = prev_chan.len();
+			let left_win_start = left_win_start as usize;
+			let right_win_start = right_win_start as usize;
+			let right_win_end = right_win_end as usize;
 
 			// Then do the actual overlap_add
-			for j in 0 .. plen {
-				chan[j + left_win_start as usize] =
-					chan[j + left_win_start as usize] * win_slope[j] +
-					prev_chan[j] * win_slope[plen as usize - 1 - j];
-  			}
+			// Set up iterators for all the variables
+			let range = {
+				let start = left_win_start;
+				let end = left_win_start + plen;
+				start..end
+			};
 
-  			// and populate the future previous half
-			let mut future_prev_half = Vec::with_capacity(
-				(right_win_end - right_win_start) as usize);
-			for i in right_win_start as usize .. right_win_end as usize {
-				future_prev_half.push(chan[i]);
+			let prev = prev_chan[0..plen].into_iter();
+
+			let (lhs, rhs) = {
+				let win_slope = &win_slope[0..plen];
+				(win_slope.iter(), win_slope.iter().rev())
+			};
+
+			for (((v, lhs), prev), rhs) in chan[range].iter_mut().zip(lhs).zip(prev).zip(rhs) {
```

Comment:
I'm not happy about this change *at all*. It's much much less clear what's going on here compared to the version with indices. But I presume it's faster. So good to go :+1: .

Reply:
We could use [itertools](https://docs.rs/itertools/0.7.8/itertools/) and it's multizip macro to simplify this, but that would add another dependency.

Reply:
Something like
```rust
for (v, lhs, prev, rhs) in izip!(chan[range], lhs, prev, rhs) {}
```

Reply:
That looks nicer but my wish to keep the dependencies of lewton lean prevails :).